### ISSUE TITLE
Stuck at Resolvers.get

### ISSUE THREAD

**Issue by pq** [dart-lang/sdk#25135]
_From @eernstg on November 3, 2015 14:57_

In some cases the `get` method on an instance of `Resolvers` from 'resolvers.dart' gets stuck, in the sense that the next statement after `await` never gets executed:

```dart
  Resolvers resolvers = ..;
  Resolver resolver = await resolvers.get(myTransform);
  // not reached
```
The problem is easy to reproduce:
```
  git clone 'https://github.com/dart-lang/reflectable'
  cd reflectable/tool
  git co 64be223a843f31b799b1170de4fe786e0468c32a
  ./bottest
```

This will end in a state where the `pub build --mode=debug test` process is stuck. The actual location where it is stuck is at line 2741 in 'lib/src/transformer_implementation.dart'. Note that the stuck state may hereafter be reproduced much faster by running `pub build --mode=debug test` directly in the 'test_reflectable' subdirectory (`cd ../test_reflectable` in the above setting).

We have noticed that the problem disappears when we use a dependency_override such that the version of the analyzer is 0.26.1+14 rather than 0.26.2. Because of this fact, we have reported the issue here as well as on the 'code_transformers' repo: https://github.com/dart-lang/code_transformers/issues/29.

Update: Reflectable has a new commit doing the dependency override on the analyzer that prevents the above problem, so the `git clone` above must fetch commit 64be223a843f31b799b1170de4fe786e0468c32a rather than the most recent one. I edited the text above to reflect this fact.

_Copied from original issue: dart-lang/analyzer_cli#73_

**Comment by eernstg** [comment]
Please reopen this: Reflectable cannot allow newer versions of the analyzer than 0.26.1+14 before this gets fixed.

**Comment by pq** [comment]
> Please reopen this: Reflectable cannot allow newer versions of the analyzer than 0.26.1+14 before this gets fixed.

It's still open!  Just moved repos.

**Comment by eernstg** [comment]
I created this as an 'analyzer_cli'  issue (Nov 3rd), but later I discovered that it should have been categorized as 'sdk' with 'area-analyzer', and I submitted a very similar issue there (Nov 24th) at https://github.com/dart-lang/sdk/issues/25032. Now this issue has moved from analyzer_cli to sdk (cf. 'Copied from..' above). I think one of them should be closed, marking it as a duplicate of the other.
